Prepend group-names to display-name

pabzm opened the pull request:
This is not optimal handling, but the most appropriate one as long as we don't actually support groups in addresss-lists. This way users can at least see the group's display-name. And we don't strip text that might be relevant to spot abusive emails.

Previously group-names were just removed, which makes it harder to spot such abusive emails.

pabzm committed:
Prepend group-names to display-name

This is not optimal handling, but the most appropriate one as long as we don't actually support groups in
addresss-lists. This way users can at least see the group's display-name. And we don't strip text that might be relevant
to spot abusive emails.

Previously group-names were just removed, which makes it harder to spot such abusive emails.

pabzm commented:
@alecpl Please note that this PR needs some more work so that the extended name gets shown in the HTML, but I would like to have some feedback before I go on with this.

alecpl commented:
Makes sense. Considering that even the most recent RFC about it is internally inconsistent, this seems the way to handle this issue. I'm not sure what more work you plan exactly.

pabzm commented:
Ok, thank you!

Currently https://github.com/roundcube/roundcubemail/blob/fbed38e2a9033542d8f593f4167d61bcdc5f146c/program/actions/mail/index.php#L1376 prevents the display-name from header values like `Test <attacker@example.org>:<admin@example.net>` are shown (to prevent phishing), thus swallowing the effort to not hide information. I'll try to make it show a warning instead.

pabzm committed:
Show full details with warning icon in case of phishing suspicion

Previously the display-name was stripped, but we should rather show it
so people can judge themselves. We insert a
warning icon with a help text to draw the attention to it.

pabzm committed:
Show homograph-warning-icon before email address, unify warning wording

This moves the warning icon that is triggered by the homograph check from the generic "notification area" (between
headers and body) to the header area, before the address that the warning is referring to.
The previous warning left it unclear which address was found to be problematic, which now is obvious.

Additionally there's now a test to check for these warnings to show up in the DOM.

pabzm commented:
I propose to show the homograph-triggered icon also before the address that it refers to, like this:

<img width="656" height="242" alt="image" src="https://github.com/user-attachments/assets/330d124c-9fa3-4c7b-82cb-4d67ade19557" />

The title text is be the same for both trigger-reasons to keep it simple.

-----

In print-mode I propose to not show the warnings, since there's no interaction possible (except for copy&paste, which I would assume that people don't do from the print-mode):


<img width="651" height="299" alt="Screenshot 2025-08-20 at 10 14 50" src="https://github.com/user-attachments/assets/c44d95d4-19fb-4ee9-a43e-d863ab3b60b5" />

pabzm commented:
@alecpl If you have any further comment please let me know in the next days. Else I'd merge this in ~1 week.

pabzm commented:
@alecpl I'll merge this in a few days unless you object
